//chat proto file

syntax = "proto3";

package grpc;

message NewNodeReq{
	int32 n_id = 1;
	int32 port = 2;
}

message FindRRequest{
  string roomname = 1;
}

message FindRResponse{
  string port = 1;
}

message JoinChatRequest{
  string roomname = 1;
  string password = 2;
  string nickname = 3;
}

message CreateChatRequest{
  string roomname = 1;
  string password = 2;
  string nickname = 3;
}

message First{
  string roomname = 1;
  string nickname = 2;
}

message Note{
  string roomname = 1;
  string nickname = 2;
  string message  = 3;
}

message EmptyResponse{}

message JoinResponse{
   string state = 1;
   int32 Port   = 2;
}

message QuitRequest{
  string roomname = 1;
  string nickname = 2;
}


service ChatSServer{
 rpc AddNewNode(NewNodeReq) returns(EmptyResponse){}
 rpc FindResponsible(FindRRequest) returns(FindRResponse){}
 rpc JoinChat(JoinChatRequest) returns(JoinResponse){}
 rpc CreateChat(CreateChatRequest) returns(JoinResponse){}

 rpc ReceiveMessage(First) returns(stream Note){}
 rpc SendMessage(Note) returns(EmptyResponse){}
 rpc Quit(QuitRequest) returns(EmptyResponse){}
}
